### ISSUE TITLE
Start mulitiple apps failure

### ISSUE THREAD

**Issue by ziyunxiao** [openanalytics/shinyproxy#415]
Hello,

My environment is Shinyproxy 3.0.1, running on Docker Swarm. When the user opens the home page and tries to open more than one app, some of the apps will fail to start.  I did not see this behaviour in version 2.6.0. 

Front end error: Failed to start app APP_NAME, "try again" button. Even I click the button, the app keeps failing. 

The error in backend logs:
```
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 2023-04-18 00:32:23.786  WARN 1 --- [pool-3-thread-2] e.o.containerproxy.service.ProxyService  : [user=xxxx@example.com proxyId=3ecd6852-5946-49e4-b563-2f5514b887fe specId=app_name] Proxy failed to start
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    |
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | eu.openanalytics.containerproxy.ProxyFailedToStartException: Container with index 0 failed to start
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at eu.openanalytics.containerproxy.backend.AbstractContainerBackend.startProxy(AbstractContainerBackend.java:116) ~[containerproxy-1.0.1.jar!/:1.0.1]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at eu.openanalytics.containerproxy.service.ProxyService.startOrResumeProxy(ProxyService.java:455) ~[containerproxy-1.0.1.jar!/:1.0.1]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at eu.openanalytics.containerproxy.service.ProxyService.lambda$startProxy$4(ProxyService.java:279) ~[containerproxy-1.0.1.jar!/:1.0.1]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at eu.openanalytics.containerproxy.service.ProxyService.lambda$action$11(ProxyService.java:599) ~[containerproxy-1.0.1.jar!/:1.0.1]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:576) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/java.lang.Thread.run(Thread.java:1623) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | Caused by: eu.openanalytics.containerproxy.ContainerFailedToStartException: Docker swarm container failed to start
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at eu.openanalytics.containerproxy.backend.docker.DockerSwarmBackend.startContainer(DockerSwarmBackend.java:221) ~[containerproxy-1.0.1.jar!/:1.0.1]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at eu.openanalytics.containerproxy.backend.AbstractContainerBackend.startProxy(AbstractContainerBackend.java:108) ~[containerproxy-1.0.1.jar!/:1.0.1]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	... 8 common frames omitted
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | Caused by: java.lang.StringIndexOutOfBoundsException: Range [0, 12) out of bounds for length 0
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:55) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:52) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckFromToIndex(Preconditions.java:112) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/jdk.internal.util.Preconditions.checkFromToIndex(Preconditions.java:349) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4727) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at java.base/java.lang.String.substring(String.java:2830) ~[na:na]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at eu.openanalytics.containerproxy.backend.docker.DockerSwarmBackend.calculateTarget(DockerSwarmBackend.java:233) ~[containerproxy-1.0.1.jar!/:1.0.1]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at eu.openanalytics.containerproxy.backend.AbstractContainerBackend.setupPortMappingExistingProxy(AbstractContainerBackend.java:248) ~[containerproxy-1.0.1.jar!/:1.0.1]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	at eu.openanalytics.containerproxy.backend.docker.DockerSwarmBackend.startContainer(DockerSwarmBackend.java:217) ~[containerproxy-1.0.1.jar!/:1.0.1]
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    | 	... 9 common frames omitted
shiny_shiny.1.nqv39mrpnkj6g2xoa1rieauta@lxvpelastsrch01.example.com    |
```

Can someone help take a look?

Thanks,

Robin

**Comment by HaraldWilhelmi** [comment]
We run into something similar after upgrading to 3.0.1. However we got as root cause "Bind for 0.0.0.0:20001 failed: port is already allocated".

In our case the issue was caused by old processes, which survived the restart of the application.
After killing two processes (docker-proxy, R), which where older than the last restart of the
service our issue vanished. Interestingly the docker-proxy blocked the port, but the R obviously
continued to report the blocked port even after we killed the docker-proxy.

Maybe it is worth to check for old R/docker-proxy processes too?
